src/views/tutorial-view/utils/variants/context.tsx

```diff
@@ -0,0 +1,52 @@
+import {
+	createContext,
+	useState,
+	useContext,
+	useMemo,
+	ReactNode,
+	Dispatch,
+	SetStateAction,
+	useEffect,
+} from 'react'
+import { TutorialVariant } from './types'
+
+type VariantContextValue = {
+	activeVariant?: TutorialVariant
+	setActiveVariant?: Dispatch<SetStateAction<TutorialVariant>>
+}
+
+interface VariantProviderProps {
+	children: ReactNode
+	variant?: TutorialVariant
+}
+
+export function useVariants(): VariantContextValue {
```

Comment:
```suggestion
export function useVariant(): VariantContextValue {
```
Maybe not make this plural to make it clear that we don't support that yet.

Reply:
👍 I could see the case for either! I'd lean slightly towards `useVariant`, but def not blocking 👍